.github/workflows/python-app.yml

```diff
@@ -39,15 +40,25 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install graphviz
-        # Note: Also need to add ffmpeg to run numpy integration tests, but omitting for time
-        run: sudo apt-get install -y graphviz
-      - name: Install dependencies
+      - uses: actions/cache@v3
+        #id: pipcache
+        with:
+          path:
+            - "/home/runner/.cache/pip"
+            - "/opt/hostedtoolcache/linea" #/opt/hostedtoolcache/linea/lib/python3.9/site-packages 
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install pip dependencies
+        #if: steps.pipcache.outputs.cache-hit != 'true'
+        run: |
+          pip install wheel && python -m pip install --user -r requirements.txt
+      - name: Install linea
         run: |
-          pip install -r requirements.txt && python setup.py install && rm -rf build
+          python -m pip install --user -r requirements.txt && python setup.py install && rm -rf build
```

Comment:
ya pytest catches the build folder and gives all sorts of errors. the build is created because we're using python setup.py install to install linea (doing this instaed of -e ".[dev]" since it is the dev install)